examples/coap/src/oluru.rs

```diff
@@ -0,0 +1,269 @@
+//! An owned heapless container with capacity N that maintains order both through properties of its
+//! entries and by time of access -- a mix between a priority queue and an LRU cache.
+//!
+//! This was inspired by the [`uluru`](https://github.com/servo/uluru) crate. Adapting that crate
+//! to support priority levels proved to be impractical -- and many properties have changed since.
+//!
+//! See the [`OrderedPool`] documentation for details.
+//!
+//! # Terminology
+//!
+//! The ordering imposed on the cache entries is determined by a priority (see [`PriorityLevel`]).
+//! Throughout the documentation, "high" and "low" indicate priorities, whereas "small" and "large"
+//! indicate numeric values (including of priorities, where "high" corresponds to "small" and "low"
+//! to "large").
+#![deny(unsafe_code)]
+
+use arrayvec::ArrayVec;
+
+/// Required trait on [OrderedPool] entries that allows ordering.
+///
+/// Priority levels follow the conventions common with schedulers: 0 is the highest priority, and
+/// will only get evicted if the cache is full with other entries of the same priority. Larger
+/// numeric values indicate increasingly lower priority.
+pub trait PriorityLevel {
+    /// Calculate the priority of the instance
+    ///
+    /// An instance's priority level may change while being mutated; [OrderedPool] will account for
+    /// that.
+    ///
+    /// The level should not change due to global effects (or internal mutability, if shared access
+    /// is later implemented). If it does, the ordering of a OrderedPool containing it may become
+    /// arbitrary, even after the elemnt whose level changed has been removed.
+    fn level(&self) -> usize;
+}
+
+/// An owned heapless container with capacity N that maintains order both through properties of its
+/// entries and by time of access.
+///
+/// Operations that are supposed to be fast are:
+///
+/// * Finding an element by iteration
+/// * Moving that element to the front of its relevant level
+///
+/// There is no remove an item; instead, the `&mut T` of [`Self::lookup`] can be replaced with a
+/// low-priority placeholder value. (In fact, future iterations may require that such a value
+/// exists and is Default).
+///
+/// # Usage
+///
+/// ```
```

Comment:
I have at some point, but only by copying the module out to the playground or similar manual action.

Reply:
I don't think that it there should be this annotation here, because that would indicate that it is *intentionally* incomplete, whereas the adequate representation of our inability to run doctests so far should be a general note in the docs that we can't run those tests yet – but I don't care enough to stall this for one more round; note added.